### ISSUE TITLE
Docker Desktop opens port 12050 by default

### ISSUE THREAD

**Issue by clarissa-adelard** [ge-high-assurance/RACK#330]
If I click on the "Open in Browser" button in Docker Desktop, I get an error response from port 12050:

> ### **Whitelabel Error Page**
> 
> This application has no explicit mapping for /error, so you are seeing this as a fallback.
> 
> Tue Mar 30 08:48:46 UTC 2021
> There was an unexpected error (type=Not Found, status=404).
> No message available

It would appear that the docker image is configured with a default port of 12050 rather than 80.

**Comment by tuxji** [comment]
I've seen this problem too.  I've googled and looked at documentation, but I haven't found out why Docker Dashboard thinks the default port is 12050 rather than 80 when both ports are published by the run command.  Maybe if the image exposes only the 80 port but not the rest of the ports, the browser still could send REST API calls to the 12050 and other ports anyway... I'll have to find out if that would work.

**Comment by tuxji** [comment]
No, changing the EXPOSE line to contain only the 80 port didn't help.  Rearranging the order of the -p options on the `docker run` command didn't help either.  The reason why the Dashboard sends the browser to the 12050 port must be because it picks the first port from all the published ports and it sorts the list alphabetically, not numerically, so it sees the 12050 port first instead of the 80 port.  

The EXPOSE line in the Docker image is completely optional; it doesn't do anything except document which ports the rack-box image's services listen to.  We already document the exact form of the `docker run` command that is needed to run the rack-box image successfully, so I've removed the EXPOSE line because it's just another place that may become out of date later.

**Comment by clarissa-adelard** [comment]
Thanks for looking into this - the behaviour of Docker Dashboard is a bit strange, but as you say, it looks as though it picks the first port from the list of all the published ports sorted alphabetically. In particular, if you click on INSPECT in Docker Dashboard, you see the list of published ports in alphabetical order, starting with 12050 and ending with 80.

Nevertheless, I think that the OPEN IN BROWSER button in Docker Dashboard should do something sensible - the current behaviour is unexpected and confusing.

One solution might be to only publish port 80 and configure the web server on the container to map URLs for the other services to the appropriate port. For example, instead of accessing the Swagger UI via port 12050, you would access it as `/swagger-ui` on port 80. 

Would something like this work?

**Comment by tuxji** [comment]
Yes, configuring the nginx web server to perform reverse port mapping from selected URLs to services' ports would allow us to publish only the 80 port (it would make the Swagger UI link in the Welcome page work without needing any JavaScript as you pointed out too).  However, we'd have to make sure all affected HTML pages and JavaScript scripts use the new URLs correctly so that kind of change will require a day or two more for effort and testing.  The ngnix web server also becomes involved in all REST calls with its own potential problems of latency, etc.  @cuddihyge Paul, have you configured SemTK with reverse port mappings in any other application?
